Tasks/DockerV2/dockerlifecycle.ts

```diff
@@ -0,0 +1,36 @@
+"use strict";
+
+import * as tl from "azure-pipelines-task-lib/task";
+import ContainerConnection from "docker-common-v2/containerconnection";
+import * as dockerCommandUtils from "docker-common-v2/dockercommandutils";
+import * as utils from "./utils";
+
+export function run(connection: ContainerConnection, outputUpdate: (data: string) => any): any {
+    let output = "";
+    let command = tl.getInput("command", true);
+    let containerName = tl.getInput("container", true);
+    let commandArguments = dockerCommandUtils.getCommandArguments(tl.getInput("arguments", false));
+
+    let containerMap = tl.getVariable("agent.containermapping");
+    if (!containerMap) {
+        throw new Error(tl.loc('ContainerNotFound', containerName));
+    }
+
+    let map = JSON.parse(containerMap);
+    let containerId = map[containerName].id;
+    if (!containerId || containerId == "") {
+        throw new Error(tl.loc('ContainerNotFound', containerName));
+    }
+
+    if (command == "start")  {
+        return dockerCommandUtils.start(connection, containerId, commandArguments, (data) => output += data).then(() => {
+            let taskOutputPath = utils.writeTaskOutput("start", output);
+            outputUpdate(taskOutputPath);
+        });
+    } else {
+        return dockerCommandUtils.stop(connection, containerId, commandArguments, (data) => output += data).then(() => {
+            let taskOutputPath = utils.writeTaskOutput("stop", output);
+            outputUpdate(taskOutputPath);
+        });
+    }
```

Comment:
I suggest changing this to `else if (command == "stop")...` with an `else` that throws on unrecognized commands. That way the assumption that the `dockerCommandMap` only uses those two values is explicit rather than implicit.

Reply:
Good idea. Changed.